### ISSUE TITLE
NOT work with legend.position = "inside".

### ISSUE THREAD

**Issue by whooooong** [YuLab-SMU/ggbreak#74]
The legend did not show up when setting legend.position = "inside".

With the most recent version of ggplot2 and ggbreak.

**Comment by veronicar239** [comment]
Yes I also noticed that the legend is missing when using theme(legend.position = c(,)) when trying to place the legend inside the plot while using scale_y_break() - screenshot 1. If I run the same code and comment out the scale_y_break() line then the legend appears as normal - screenshot 2.

data %>%
  drop_na(classKey) %>%  #drop_na(class) - still shows NA groups among departments
      group_by(Department, class) %>%
        dplyr::summarise(
          count = n()) %>% 
            arrange(desc(count)) %>% 
              head(50) %>% 
  ungroup() %>%
  mutate(class = fct_reorder(class, count, .desc = TRUE)) %>% 
  ggplot(aes(x=class, y=count, fill = Department)) +
    geom_bar(stat = "identity") +
    scale_y_break(breaks = c(1000000, 2500000)) + #scale_y_break(c(break_point, max(your_data$value))) # breaks = c(50)
    scale_fill_discrete(type = palette_secondary_6_green) +
    labs(#title = "Top 50 classes",
       x = "Class",
       y = "Occurences",
       fill = "Department") +
    theme_minimal() +
    theme(legend.position = c(.85,.31), axis.text.x = element_text(angle = 45, hjust=1))
 
screenshot 1
<img width="923" alt="Image" src="https://github.com/user-attachments/assets/60648371-b0b5-409a-ad9e-32523096b165" />

screenshot 2

<img width="844" alt="Image" src="https://github.com/user-attachments/assets/09ea2d33-e1cc-4154-ae65-3445e4a016f8" />

**Comment by GuangchuangYu** [comment]
Confirmed on the current `master` (`ggbreak 0.1.7.009`, ggplot2 4.0.3):

    ggplot(dd, aes(x, y, color = g)) + geom_point() +
      theme(legend.position = "inside", legend.position.inside = c(0.8, 0.25)) +
      scale_y_break(c(5, 15))

— no legend appears anywhere in the rendered image. Without the break, the same code draws the legend at (0.8, 0.25) inside the panel as expected.

Root cause is in how the per-subplot guides are reassembled: `ggbreak` builds the subplots and feeds them to `aplot::plot_list(..., guides = "collect", output = "patchwork") & legendpos`. The collected-guide machinery (patchwork / `&`) understands `legend.position = "right" | "left" | "top" | "bottom" | "none"`, but does not honour `legend.position.inside`; the result is that the legend key set is collected, placed somewhere that has no patch of pixel where the user asked for it, and disappears.

Workarounds on the current dev release:

    p + theme(legend.position = "right")   # works
    p + theme(legend.position = "bottom")  # works
    p + guides(color = "none")            # hide legend

A real fix would need us to bypass the collected-guide path when `legend.position = "inside"` and instead draw the legend at the requested `legend.position.inside` coordinates ourselves. That's substantial new code; I'd like to first see how many users actually want "inside" with a broken plot (and so on which axes) before investing in it. Closing as `not planned` for now — please reopen if you want this prioritised, and ideally describe the workflow you have in mind so I can sanity-check the workaround.
